operator/pkg/reconciliation/reconcile_racks.go

```diff
@@ -194,6 +195,9 @@ func (rc *ReconciliationContext) CheckRackPodTemplate() result.ReconcileResult {
 
 		if needsUpdate {
 
+			rc.Recorder.Eventf(rc.Datacenter, "Normal", "UpdatingRack",
```

Comment:
What does everyone think about defining an enum for the "reason" argument?  The docs say:

> // 'type' of this event, and can be one of Normal, Warning. New types could be added in future
> // 'reason' is the reason this event is generated. 'reason' should be short and unique; it
> // should be in UpperCamelCase format (starting with a capital letter). "reason" will be used
> // to automate handling of events, so imagine people writing switch statements to handle them.
> // You want to make that easy.

( https://godoc.org/k8s.io/client-go/tools/record#EventRecorder )

Making an enum for the reasons would allow us to easily audit all events at a high-level.

Reply:
You are right, we should make these constants. The reason values I have now are intended to be what is described in the docs. Do you want changes to the actual reason values?

Reply:
No, the values seem appropriate to me.

Reply:
If there is any feedback, it's just to be overly pedantic on verb tense.  If we have completely finished an action, make sure we use past tense like "CreatedResource", and if we are about to do something or we are currently doing it, let's make sure it's a continuous tense like "DeletingStuckPod".  The stuff like "SetPodAsSeed" is neither past tense nor continous, and is slightly ambiguous.

This is uber-pedantic, but it would remove the possibility of someone getting confused.

Reply:
Also, "UpateRackLabelsOnResource" has a typo.  I broke the existing values into groups based on my grammar mania:

Ongoing Actions:

UpdatingRack
StoppingDatacenter
DeletingStuckPod
RestartingCassandra

Completed Actions:

CreatedResource
StartedCassandra
SetPodAsSeed
UnsetPodAsSeed

Ambiguous Actions:

UpateRackLabelsOnResource
ScaleUpRack
CreateSuperuser
FinishReplaceNode
StartReplaceNode
StartCassandraAndReplaceNode
StartCassandra